src/dvclive/live.py

```diff
@@ -929,10 +929,17 @@ def post_data_to_studio(self):
             self._studio_queue = queue.Queue()
 
             def worker():
+                error_occurred = False
                 while True:
                     item, data = self._studio_queue.get()
-                    post_to_studio(item, "data", data)
-                    self._studio_queue.task_done()
+                    try:
+                        if not error_occurred:
+                            post_to_studio(item, "data", data)
+                    except Exception:
+                        logger.exception("Failed to post data to studio")
+                        error_occurred = True
+                    finally:
+                        self._studio_queue.task_done()
```

Comment:
Can we use `thread.is_alive()` to check and never push the data (or some other `Event`) instead of keeping a thread alive?

Reply:
there can be still a race condition - thread is just still alive (but about to exit), we do a check, we push data and then  `queue.join()` hangs. I would rather to something dumb and simple (even in this case easily I missed some issues - multithreading is hard)